pkg/domain/filters/containers.go

```diff
@@ -18,9 +19,20 @@ import (
 func GenerateContainerFilterFuncs(filter string, filterValues []string, r *libpod.Runtime) (func(container *libpod.Container) bool, error) {
 	switch filter {
 	case "id":
-		// we only have to match one ID
+		hexRegex := regexp.MustCompile("[^0-9a-fA-F]")
```

Comment:
this should be a global variable and use the `"github.com/containers/storage/pkg/regexp"` package instead.